libraries/consul_acl.rb

```diff
@@ -90,10 +90,35 @@ def configure_diplomat
       end
 
       def up_to_date?
-        old_acl = Diplomat::Acl.info(new_resource.to_acl['ID'], nil, :return)
-        return false if old_acl.nil? || old_acl.empty?
-        old_acl.first.select! { |k, _v| %w(ID Type Name Rules).include?(k) }
-        old_acl.first == new_resource.to_acl
+        retry_block(attempts: 3, sleep: 0.5) do
+          old_acl = Diplomat::Acl.info(new_resource.to_acl['ID'], nil, :return)
+          return false if old_acl.nil? || old_acl.empty?
+          old_acl.first.select! { |k, _v| %w(ID Type Name Rules).include?(k) }
+          old_acl.first == new_resource.to_acl
+        end
+      end
+
+      def retry_block(opts = {}, &_block)
+        opts = {
+          attempts: 3, # Number of attempts
+          sleep:    0, # Seconds to sleep between attempts
+        }.merge(opts)
+
+        attempts = 1
```

Comment:
I would suggest renaming `attempts` to something like `count`, or `opt[:attempts]` to `opt[:max_attempts]` in order to avoid the confusion between these two variables.